core/src/banking_stage/transaction_scheduler/in_flight_tracker.rs

```diff
@@ -0,0 +1,142 @@
+use {
+    super::thread_aware_account_locks::ThreadId,
+    crate::banking_stage::scheduler_messages::TransactionBatchId, std::collections::HashMap,
+};
+
+/// Tracks the number of transactions that are in flight for each thread.
+pub struct InFlightTracker {
+    num_in_flight_per_thread: Vec<usize>,
+    cus_in_flight_per_thread: Vec<u64>,
+    batches: HashMap<TransactionBatchId, BatchEntry>,
```

Comment:
Looks it doesn't really do much with stored `BatchEntry` here. (Maybe it is used somewhere else) But here it looks like the HashMap only used to check id BatchId already in-flight? If so, why store the `BatchEntry`?

Per the other comment, if the duplicated batch id checking is done elsewhere, may not need the entire `batches`, which makes the Tracker's function clearer - just tracking the number of txs and number of CUs.

Reply:
The batch entries are used for a couple reasons:
1. no need to re-calculate number of CUs in the batch when we complete it
2. no possibility the caller sends back the wrong num_transactions/total_cus for a complete batch

Reply:
Ah, I missed the part to remove batch's counts when compete. Makes sense to cache it.